base/leaky_bucket.go

```diff
@@ -344,7 +344,8 @@ func (b *LeakyBucket) wrapFeedForDeduplication(args sgbucket.FeedArguments, dbSt
 
 		// the buffer to hold tap events that are candidates for de-duplication
 		deDupeBuffer := []sgbucket.FeedEvent{}
-
+		ticker := time.NewTicker(deDuplicationTimeoutMs)
```

Comment:
This looks like similar logic to #4479 which I've just reworked to prevent the loop from spinning when idle.

We might want to apply similar changes here.

Reply:
Changes have been committed.